.pipelines/e2e-tests-template.yml

```diff
@@ -0,0 +1,122 @@
+              echo "##vso[task.setvariable variable=MIC_VERSION]${MIC_VERSION}"
+              export NMI_VERSION="${IMAGE_VERSION}"
+              echo "##vso[task.setvariable variable=NMI_VERSION]${NMI_VERSION}"
+              export IDENTITY_VALIDATOR_VERSION="${IMAGE_VERSION}"
+              echo "##vso[task.setvariable variable=IDENTITY_VALIDATOR_VERSION]${IDENTITY_VALIDATOR_VERSION}"
+
+              az acr login -n $(REGISTRY_NAME)
+              make image-mic image-nmi image-identity-validator
+              make push-mic push-nmi push-identity-validator
+            displayName: "Build and push MIC, NMI and identity-validator"
+
+          - script: |
+              echo "Installing aks-engine..."
+              curl -LO https://github.com/Azure/aks-engine/releases/download/$(AKS_ENGINE_VERSION)/aks-engine-$(AKS_ENGINE_VERSION)-linux-amd64.tar.gz
+              tar -zxvf aks-engine-$(AKS_ENGINE_VERSION)-linux-amd64.tar.gz
+              sudo mv aks-engine-$(AKS_ENGINE_VERSION)-linux-amd64/aks-engine /usr/local/bin/
+              echo
+
+              echo "Installing kubectl..."
+              curl -LO https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/amd64/kubectl
+              chmod +x kubectl
+              sudo mv kubectl /usr/local/bin/
+            displayName: "Install tools"
+
+          - script: |
+              # Generate SSH keypair
+              echo -e 'y\n' | ssh-keygen -f id_rsa -t rsa -N '' > /dev/null
+              export SSH_PUBLIC_KEY="$(cat id_rsa.pub)"
+
+              # Generate resource group name
+              export RESOURCE_GROUP="aad-pod-identity-e2e-$(openssl rand -hex 6)"
+              echo "##vso[task.setvariable variable=RESOURCE_GROUP]${RESOURCE_GROUP}"
+
+              envsubst < test/e2e/cluster_configs/$(CLUSTER_CONFIG).json > kubernetes.json
+              cat kubernetes.json
+            displayName: "Generate API model"
+
+          - script: |
+              aks-engine deploy \
+                --api-model kubernetes.json \
+                --location $(LOCATION) \
+                --subscription-id $(SUBSCRIPTION_ID) \
+                --client-id $(AAD_CLIENT_ID) \
+                --client-secret $(AAD_CLIENT_SECRET)
+
+              export KUBECONFIG="$(pwd)/_output/${RESOURCE_GROUP}/kubeconfig/kubeconfig.$(LOCATION).json"
+              echo "##vso[task.setvariable variable=KUBECONFIG]${KUBECONFIG}"
+
+              # Sleep for 120 seconds to wait for nodes and pods to become ready
+              sleep 120
+              kubectl wait --for=condition=ready node --all
+              kubectl wait pod -n kube-system --for=condition=Ready --all
+              kubectl get nodes -owide
+              kubectl cluster-info
+
+              # Install gatekeeper
+              kubectl apply -f https://raw.githubusercontent.com/open-policy-agent/gatekeeper/master/deploy/gatekeeper.yaml
```

Comment:
Why are we installing gatekeeper?

Reply:
Didn't realize we install gatekeeper during the test run